libp2p/protocols/pubsub/gossipsub.nim

```diff
@@ -942,9 +950,9 @@ proc punishPeer(g: GossipSub, peer: PubSubPeer, topics: seq[string]) =
     # ensure we init a new topic if unknown
     let _ = g.topicParams.mgetOrPut(t, TopicParams.init())
     # update stats
-    var tstats = g.peerStats[peer].topicInfos.getOrDefault(t)
-    tstats.invalidMessageDeliveries += 1
-    g.peerStats[peer].topicInfos[t] = tstats
+    g.peerStats.withValue(peer, stats):
```

Comment:
unlike the previous version, this will not add a _new_ entry to the table if it's missing.. intended?

Reply:
yep it's fine

Reply:
actually I see your point for this specific line, let me review it.

Reply:
fixed, nice catch, the table was populated on graft but it was very possible that we received and we definitely needed to score prior to that